### DISCUSSION TITLE
Is it possible to add a linear regression line?

### DISCUSSION THREAD

**Discussion by mhkeller** [perspective-dev/perspective#2602] [Q&A] [answered]
I have a feeling this may be out of scope but I was curious if anyone has had success in creating a linear regression line using column expressions.

**Comment by texodus** [comment, marked as answer]
It is not ergonomic to calculate a linear regression in Perspective currently, though this can be done with `vlookup()` and a `for` loop if all of your `Table`'s indices can be easily iterated over. It is also difficult to plot one if you calculate the parameters independently due to bugs/misbehavior - e.g. `"Y Line"` and `"Y Scatter"` seem to erroneously set `float` columns as category axes when used in the `Group By` position, though it is easy to calculate e.g. `a * "Sales" + b` linear regression from the column `"Sales"` with known `a` and `b`.

This is not out of scope though. We can conquer this in a number of ways, but e.g. `Candlestick` charts calculate a moving average window independently already, which may be a good pattern to follow for this type of common chart analytics.

**Reply by mhkeller** [reply]
Thanks for the quick answer.
